metricflow/model/transformations/identifiers.py

```diff
@@ -25,7 +25,7 @@ def transform_model(model: UserConfiguredModel) -> UserConfiguredModel:  # noqa:
                         continue
 
                     for identifier in data_source.identifiers:
-                        if sub_identifier.ref == identifier.name.element_name:
+                        if sub_identifier.ref == identifier.name:
```

Comment:
I think it makes sense to delete `sub_identifier.ref` and replace its usage with `sub_identifier.reference.element_name`.